assets/src/components/detours/detourDropdown.tsx

```diff
@@ -0,0 +1,82 @@
+import React from "react"
+import { DropdownItem, DropdownMenu } from "../map/dropdown"
+import { Route, RoutePattern, Shape } from "../../schedule"
+import { Popup } from "react-leaflet"
+import { PointTuple } from "leaflet"
+import { LatLngLiteral } from "leaflet"
+
+export interface StartDetourProps {
+  routeName: string
+  routeDescription: string
+  routeOrigin: string
+  routeDirection: string
+  shape: Shape
+  center: LatLngLiteral
+  zoom: number
+}
+
+export interface DetourDropdownProps {
+  setShouldShowPopup: (newValue: boolean) => void
+  routePatternForVehicle: RoutePattern | null
+  route: Route | null
+  onStartDetour?: (props: StartDetourProps) => void
+  onClick: (props: StartDetourProps) => void
+  center: LatLngLiteral
+  zoom: number
+}
+
+export const DetourDropdown = ({
+  routePatternForVehicle,
+  route,
+  onClick,
+  center,
+  zoom,
+}: DetourDropdownProps) => {
+  // This offset is here because, due to a limitation of Leaflet
+  // popups, we weren't able to render the popup at the bottom-right
+  // corner of the marker, where it's supposed to go. This effectively
+  // renders it centered and above the marker, and then uses the
+  // offset to reposition it to the bottom-right corner.
+  const dropdownOffset: PointTuple = [140, 97]
+
+  if (!routePatternForVehicle || !route) {
+    return <></>
+  }
+
+  const routeId = routePatternForVehicle.routeId
+
+  const routeDescription = routePatternForVehicle.headsign
+
+  const routeOrigin = routePatternForVehicle.name
+
+  const routeDirection =
+    route.directionNames[routePatternForVehicle.directionId]
+
+  const shape = routePatternForVehicle.shape
+
+  if (!routeDescription || !shape) {
+    return <></>
+  }
+
+  return (
+    <Popup className="c-dropdown-popup-wrapper" offset={dropdownOffset}>
+      <DropdownMenu>
+        <DropdownItem
+          onClick={() => {
+            onClick({
+              routeName: routeId,
```

Comment:
**issue:** be careful not to mix up route name and ID - they're mostly the same, but there are examples where they aren't (for instance the SL1 has route ID 741).

Reply:
Thankfully there is an application-wide context providing all of the route objects, so you can use that if you need to do the mapping.

Reply:
Ahhhhh I was wondering about that.

Reply:
How do I actually get the route name out of the routePattern?

When I console.logged `routePatternForVehicle` for an SL2 bus, it didn't have any fields whose value was `"SL2"`

Reply:
It's the route, not the route pattern, that has the name. You'll need to key off of `routePatternForVehicle.routeId` to look up the appropriate route.

Reply:
Ooh actually this was a pretty easy fix, since I had access to the `route` in this context as well: https://github.com/mbta/skate/pull/2377/commits/386996b161b5d6f55b2211cb2af70050a2e66786